src/aap_eda/api/views/credential_input_source.py

```diff
@@ -0,0 +1,243 @@
+            status.HTTP_201_CREATED: OpenApiResponse(
+                serializers.CredentialInputSourceSerializer,
+                description="Return the new Credential Input Source.",
+            ),
+        },
+    )
+    def create(self, request, *args, **kwargs):
+        return self._create_credential_input_source(request, request.data)
+
+    @extend_schema(
+        description="List all Credential Input Sources",
+        responses={
+            status.HTTP_200_OK: OpenApiResponse(
+                serializers.CredentialInputSourceSerializer(many=True),
+                description="Return a list of Credential Input Sources.",
+            ),
+        },
+    )
+    def list(self, request, *args, **kwargs):
+        objects = self.get_queryset()
+        objects = self.filter_queryset(objects)
+
+        serializer = serializers.CredentialInputSourceSerializer(
+            objects, many=True
+        )
+        result = self.paginate_queryset(serializer.data)
+
+        return self.get_paginated_response(result)
+
+    @extend_schema(
+        description="Partial update of an Credential Input Source",
+        request=serializers.CredentialInputSourceUpdateSerializer,
+        responses={
+            status.HTTP_200_OK: OpenApiResponse(
+                serializers.CredentialInputSourceSerializer,
+                description=(
+                    "Update successful. Return an updated "
+                    "credential input source."
+                ),
+            )
+        },
+    )
+    def partial_update(self, request, *args, **kwargs):
+        obj = self.get_object()
+        data = request.data
+
+        data["metadata"] = inputs_to_store_dict(
+            data.get("metadata", {}), obj.metadata
+        )
+
+        serializer = serializers.CredentialInputSourceUpdateSerializer(
+            obj, data=data, partial=True
+        )
+        serializer.is_valid(raise_exception=True)
+
+        metadata = serializer.validated_data.get("metadata")
+        if metadata or metadata == {}:
+            serializer.validated_data["metadata"] = inputs_to_store(
+                metadata,
+                obj.metadata,
+            )
+
+        old_data = model_to_dict(obj)
+        for key, value in serializer.validated_data.items():
+            setattr(obj, key, value)
+
+        with transaction.atomic():
+            obj.save()
+            check_related_permissions(
+                request.user,
+                serializer.Meta.model,
+                old_data,
+                model_to_dict(obj),
+            )
+
+        obj.refresh_from_db()
+        return Response(
+            serializers.CredentialInputSourceSerializer(obj).data,
+        )
+
+    @extend_schema(
+        description="Delete an credential input source by id",
+        responses={
+            status.HTTP_204_NO_CONTENT: OpenApiResponse(
+                None, description="Delete successful."
+            )
+        },
+        parameters=[
```

Comment:
Do we need this `force` para? Did see it in the function definition

Reply:
Done